s/core_test.go

```diff
@@ -205,3 +205,50 @@ func TestStartNestedSupervisors(t *testing.T) {
 		)
 	})
 }
+
+func TestStartFailedChild(t *testing.T) {
+	parentName := "root"
+	b0n := "branch0"
+	b1n := "branch1"
+
+	cs := []c.ChildSpec{
+		WaitDoneChild("child0"),
+		WaitDoneChild("child1"),
+		WaitDoneChild("child2"),
+		FailStartChild("child3"),
+	}
+
+	b0 := s.New(b0n, s.WithChildren(cs[0], cs[1]))
+	b1 := s.New(b1n, s.WithChildren(cs[2], cs[3]))
+
+	events, err := ObserveSupervisor(
+		context.TODO(),
+		parentName,
+		[]s.Opt{
+			s.WithSubtree(b0),
+			s.WithSubtree(b1),
+		},
+		func(em EventManager) {},
+	)
+
+	assert.Error(t, err)
+
+	AssertExactMatch(t, events,
+		[]EventP{
+			// start children from left to right
+			ProcessStarted("root/branch0/child0"),
+			ProcessStarted("root/branch0/child1"),
+			ProcessStarted("root/branch0"),
+			ProcessStarted("root/branch1/child2"),
+			// we get the failing start
+			ProcessFailed("root/branch1/child3"),
+			// immediately stop started children in the reverse order
```

Comment:
Something is not quite obvious to me: will everything stop, or just what started before this point? What if there is a good `child4`?

Reply:
If `child4` is started before the failing child, it would have been started and then stopped. If it is planned to be started after `child3`, but `child3` fails to start, then, `child4` will never start. The procedure is to abort the whole supervision tree when there is an error.
 
I'm going to add a comment here in this test so it is clear

Reply:
Got it! yes, a note to that effect would be great. In the test, can you assert that condition (with a `// note to that effect` for the reader)? It is totally reasonable, it's just that the language elsewhere (in the PR) made me think otherwise.